### PR TITLE
Add create release branch action

### LINKED ISSUE #27: Add New Branch Action
- Uses the GitHub App Credentials
- Inputs are:
	- base ref
	- new branch name 
	- purls file
	- sbom location

---
- Creates branch
- Creates a new sbom lite file
- Creates new silk asset group
- Sets release workflow's silk asset group 
- Commits and pushes

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #27

Tested with: https://github.com/blink1073/winkerberos/actions/runs/11235828228/job/31299944398

### CODE REVIEW

File: create-branch/create-branch.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+set -x
+SERIAL=$(jq -r '.serialNumber' ${TMP_SBOM})
+TIMESTAMP=$(jq -r '.metadata.timestamp' ${TMP_SBOM})
+rm ${TMP_SBOM}
+
+cat ${SBOM_FILE_PATH}
+echo "Replace the values in the existing sbom."
+cat <<< "$(jq --indent 4 '.serialNumber = "'${SERIAL}'"' ${SBOM_FILE_PATH})" > ${SBOM_FILE_PATH}
+cat <<< "$(jq --indent 4 '.metadata.timestamp = "'${TIMESTAMP}'"' ${SBOM_FILE_PATH})" > ${SBOM_FILE_PATH}
+cat ${SBOM_FILE_PATH}
```
Comment (alcaeus) [inline create-branch/create-branch.sh:68]: Not sure I follow exactly what's happening here. We fetch the SBOM for the new branch, then bump the serial number in the SBOM file path - why is that?

File: create-branch/create-branch.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+echo "Update the workflow with the silk asset group and evergreen project."
+sed -i 's/SILK_ASSET_GROUP:.*/SILK_ASSET_GROUP: '${SILK_GROUP}'/' ${RELEASE_WORKFLOW_PATH}
+sed -i 's/EVERGREEN_PROJECT:.*/EVERGREEN_PROJECT: '${EVERGREEN_PROJECT}'/' ${RELEASE_WORKFLOW_PATH}
```
Comment (alcaeus) [inline create-branch/create-branch.sh:72]: I think this relates to my earlier comment about the asset group prefix vs. hardcoded asset name. I'd personally change the release tooling to use the same prefix logic for the silk asset group.
As for evergreen project name, we don't use that in our PHP release workflow, but I can see how other drivers may need it. Wouldn't it make more sense to use the same `<prefix>-<branch-name>` for those names so we don't need to update this information for each release branch?

File: README.md [hunk 3]
```diff
@@ -255,11 +255,44 @@ By default, all files in the S3 directory are uploaded. When the `dry_run` input
...
+It will create a new Silk Asset Group, update the SBOM-lite file,
+update the ``SILK_ASSET_GROUP`` and ``EVERGREEN_PROJECT`` env variables
+in the release workflow file, bump the version to a
+prerelease version, and push the changes.
+
+> [!Note]
+> You will need to wait overnight before making a release on
+> the new branch to allow Silk to be populated, so it is recommended to
+> make a minor/major release prior to creating a release branch, or create the
+> release branch at least one day before a planned release.
```
[the remark this answers is not in the document]
Comment (blink1073) [reply README.md:270]: Correct

File: create-branch/create-branch.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    "code_repo_url": "https://github.com/${OWNER_REPO}",
+    "branch": "${BRANCH}",
+    "metadata": {
+        "sbom_lite_path": "${SBOM_FILE_PATH}"
+    },
+    "file_paths": [],
+    "asset_id": "$SILK_GROUP"
+}
+EOF
+)
```
[the remark this answers is not in the document]
Comment (blink1073) [reply create-branch/create-branch.sh:43]: This config works for all repos that I maintain, including libmonocrypt/bindings/python, but yeah, we could make it templatable if needed.

File: create-branch/create-branch.sh (hunk 1 above)
Comment (blink1073) [reply create-branch/create-branch.sh:68]: The SBOM needs to be unique per branch if I understand correctly.

File: create-branch/create-branch.sh (hunk 2 above)
Comment (blink1073) [reply create-branch/create-branch.sh:72]: We use the evergreen project name to tie the release to an Evergreen run url, which is included in the compliance report.  Doesn't each branch need its own Evergreen project?  Either way, we could make this part optional if no evergreen project is given.

**Review by alcaeus — APPROVED** [review]
LGTM if it works for Python and go. We can always make changes for other drivers if and when they are necessary 👍

**Pull request merged**